gtda/mapper/utils/_visualization.py

```diff
@@ -176,102 +167,144 @@ def replace_summary_statistic(current_hovertext, new_statistic):
-            raise ValueError(
-                "color_variable and data must have the same length.")
-        color_variable_kind = "scalars"
-    elif hasattr(color_variable, "transform"):
-        color_variable_kind = "transformer"
-    elif hasattr(color_variable, "fit_transform"):
-        color_variable_kind = "fit_transformer"
-    elif callable(color_variable):
-        color_variable_kind = "callable"
-    elif color_variable is None:
-        color_variable_kind = "none"
-    else:  # Assume color_variable is a selection of columns
-        color_variable_kind = "else"
-
-    return color_variable_kind
-
-
-def _get_node_summary_statistics(
-        data, is_data_dataframe, node_elements, summary_statistic,
-        color_variable
-        ):
-    """Calculate values of node summary statistics."""
-    color_variable_kind = _infer_color_variable_kind(color_variable, data)
-
-    if color_variable_kind == "scalars":
-        color_data = color_variable
-    elif color_variable_kind == "transformer":
-        color_data = color_variable.transform(data)
-    elif color_variable_kind == "fit_transformer":
-        color_data = color_variable.fit_transform(data)
-    elif color_variable_kind == "callable":
-        color_data = color_variable(data)
-    elif color_variable_kind == "none":
-        if is_data_dataframe:
-            color_data = data.to_numpy()
+    if hasattr(color_features, "dtype") or hasattr(color_features, "dtypes"):
+        raise ValueError("`color_features` should not be a numpy array or "
+                         "pandas dataframe.")
+    elif hasattr(color_features, "fit_transform"):
+        color_features_kind = "fit_transformer"
+    elif hasattr(color_features, "transform"):
+        color_features_kind = "transformer"
+    elif callable(color_features):
+        color_features_kind = "callable"
+    elif color_features is None:
+        color_features_kind = "none"
+    else:  # Assume color_features is a selection of columns
+        color_features_kind = "else"
+
+    return color_features_kind
+
+
+def _validate_color_kwargs(graph, data, color_data, color_features,
+                           node_color_statistic, interactive=False):
+    if node_color_statistic is None:
+        node_color_statistic = np.mean
+
+    if color_data is None:
+        color_data = np.arange(len(data))
+
+    color_data_checked = check_array(color_data, ensure_2d=False, dtype=None)
+
+    # Simple duck typing to determine whether `color_data` is likely a pandas
+    # dataframe
+    is_color_data_dataframe = hasattr(color_data, "columns")
+
+    if len(color_data) != len(data):
+        raise ValueError("`color_data` and `data` must have the same length.")
+    if not is_color_data_dataframe:
+        color_data = color_data_checked.reshape((len(color_data), -1))
+
+    # Determine whether node_color_statistic is an array of node colors
+    is_node_color_statistic_ndarray = hasattr(node_color_statistic, "dtype")
```

Comment:
Shouldn't it be `is_node_color_statistic_array`? In fact, we don't check the number of dimensions (dim?)

Reply:
I see that you resolved this, but just to be clear I mean https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html, i.e. I'm using ndarray as a synonym of numpy array.